cli/flox/src/commands/install.rs

```diff
@@ -107,12 +140,46 @@ impl Install {
                 })
             },
             Err(e @ EnvironmentSelectError::EnvNotFoundInCurrentDirectory) => {
-                // TODO: prompt for initialization here
-                bail!(formatdoc! {"
-                {e}
-
-                Create an environment with 'flox init' or install to an environment found elsewhere with 'flox install {} --dir <PATH>'",
-                self.packages.join(" ")})
+                let user_state_path = user_state_path(&flox);
+                let (lock, mut user_state) = lock_and_read_user_state_file(&user_state_path)?;
+                if user_state.confirmed_create_default_env.is_some() {
+                    bail!(formatdoc! {"
+                    {e}
+
+                    Create an environment with 'flox init' or install to an environment found elsewhere with 'flox install {} --dir <PATH>'",
+                    self.packages.join(" ")});
+                }
+                let msg = formatdoc! {"
+                    Packages must be installed into a Flox environment, which can be
+                    a user 'default' environment or attached to a directory.
+                "};
+                message::plain(msg);
+                let package_list = package_list_for_prompt(&packages_to_install)
+                    .context("must specify at least one package to install")?;
+                let (choice_idx, _) = Dialog {
+                    message: &format!(
+                        "Would you like to install {package_list} to the 'default' environment?"
+                    ),
+                    help_message: None,
+                    typed: Select {
+                        options: vec!["Yes", "No"],
+                    },
+                }
+                .raw_prompt()?;
+                let should_install_to_default_env = choice_idx == 0;
+                if !should_install_to_default_env {
+                    user_state.confirmed_create_default_env = Some(false);
+                    write_user_state_file(&user_state, &user_state_path, lock)
+                        .context("failed to save default environment choice")?;
+                    let msg = format!("Create an environment with 'flox init' or install to an environment found elsewhere with 'flox install {} --dir <PATH>'", self.packages.join(" "));
+                    message::plain(msg);
+                    return Ok(());
```

Comment:
Note: this isn't returning an error because @jennymahmoudi asked that we not show an error when the user declines to install packages to the default environment.

Reply:
We should still exit 1 even if we don't print anything